pallets/identity/src/keys.rs

```diff
@@ -884,9 +884,34 @@ impl<T: Config> Module<T> {
         ensure_custom_length_ok::<T>(perms.complexity(), MAX_PERMISSION_COMPLEXITY)?;
         ensure_custom_length_ok::<T>(perms.asset.complexity(), MAX_ASSETS)?;
         ensure_custom_length_ok::<T>(perms.portfolio.complexity(), MAX_PORTFOLIOS)?;
+        Self::ensure_no_except_perms(&perms.extrinsic)?;
         Self::ensure_extrinsic_perms_length_limited(&perms.extrinsic)
     }
 
+    // Ensures that extrinsic permissions do not use the Except variant
```

Comment:
```suggestion
    /// Ensures that extrinsic permissions do not use the Except variant.
    /// This is considered unsafe since extrinsic names can change or be replaced with newer versions
    pub fn ensure_no_except_perms(extrinsic_permissions: &ExtrinsicPermissions) -> DispatchResult {
        match extrinsic_permissions {
            SubsetRestriction::Except(_) => Err(Error::<T>::ExceptNotAllowedForExtrinsics.into()),
            SubsetRestriction::Whole => Ok(()),
            SubsetRestriction::These(pallet_permissions) => {
                for permission in pallet_permissions {
                    if let SubsetRestriction::Except(_) = permission.dispatchable_names {
                        return Err(Error::<T>::ExceptNotAllowedForExtrinsics.into());
                    }
                }
                Ok(())
            }
        }
    }
```

Reply:
There is also the case where `permission.dispatchable_names` is also a set (not covered in the suggestion above)

Reply:
As discussed, there is not further nesting needed to cover here.